### ISSUE TITLE
Standardize usage of globbing across CLI verbs

### ISSUE THREAD

**Issue by piotrpMSFT** [dotnet/cli#3409]
`dotnet build` supports globbing pattern inputs, while publish, pack, test do not. We need to decide the intended usage of globbing and standardize it across verbs.

/cc @pakrym who added globbing to `build` for thoughts on why we only added globbing in `build`

**Comment by brthor** [comment]
This will have interesting ramifications in bash where `**` for example is evaluated before being passed off to the process.

**Comment by pakrym** [comment]
We added globbing to build for performance reasons, when you do `dotnet build src\**\project.json` all lock files are read only once and incremental checks are ran once too, for solutions with many projects this brings measurable speed improvement.

**Comment by dariusjs** [comment]
@piotrpMSFT in plain english does this issue describe not being able to include dependencies in a single "dotnet pack"  like the report here https://github.com/dotnet/cli/issues/3756

It is something I am experiencing even though I follow the convention here https://docs.microsoft.com/en-us/dotnet/articles/core/tools/dotnet-pack

**Comment by piotrpMSFT** [comment]
#4569 describes how this was resolved.

**Comment by blackdwarf** [comment]
@piotrpMSFT is this still an issue? Also, with our move to MSBuild, isn't globbing standardized to whatever MSBuild is supporting?

**Comment by piotrpMSFT** [comment]
We should keep using this issue to ensure that all msbuild-using verbs are able to act on:
project paths
project directories
solution paths
solution directories

This needs to be consistent.

**Comment by piotrpMSFT** [comment]
globbing is no longer accepted as an input to CLI commands. Instead, if wanting to act on multiple projects then a sln file should be used.

**Comment by mwijnands** [comment]
But `dotnet test` does not accept a sln file.

**Comment by jeremysimmons** [comment]
@GuardRex are you saying the solution that #1834 solved could e applied to `dotnet test` ?

**Comment by GuardRex** [comment]
@jeremysimmons I think so. The situation with the `dotnet sln add` topic was that it was very unclear about globbing generally. @gforceg made a major improvement to the topic, but it was just something we were working on for that one command.

I referenced this issue because the discussion here made it sound like globbing was unsupported until that was clarified by [@]livarcocc. Since the globbing is platform-dependent and we weren't angling to take it on with our own dedicated topic, we just point the reader to the [Globbing patterns](https://en.wikipedia.org/wiki/Glob_(programming)) topic on Wikipedia for more info, where they do a nice job of calling out the platform-specific details.

Wherever globbing is supported, I think we should do something similar. I have [\[2.0\] Create CLI command preview topics (dotnet/docs 2139)](https://github.com/dotnet/docs/issues/2139) on my plate, and I should be getting on that *very* soon ... maybe even tomorrow if things go well today :pray:. I'll make an effort on this upcoming pass to shore up the globbing coverage.

**Comment by dasMulli** [comment]
@jeremysimmons fyi  there are a few workarounds:

1. you can use an msbuild project file run tests on multiple projects without relying on a platform's shell: https://github.com/dotnet/cli/issues/6572#issuecomment-300921019 (more integrated "build script": https://gist.github.com/dasMulli/69f5303aa79a8cd4060e44891c90fd2d)

2. create a `Directory.Build.props` file containing `<Project><Target Name="VSTest" /></Project>` - this is a dirty workaround to be able to `dotnet test the.sln` by adding `VSTest` target to all projects (real test projects will override the empty target through imported NuGet packages).

**Comment by jeremysimmons** [comment]
@dasMulli I could not get either option to work. I'm using XUnit, not VSTest

**Comment by CoskunSunali** [comment]
May I ask why is this issue closed? Is it already implemented, being tracked in another issue or will not be implemented at all?
